### ISSUE TITLE
Error when transitioning away from a page with an Arrow

### ISSUE THREAD

**Issue by bigonese** [shoes/shoes4#1454]
When I'm on a view with an Arrow and then `visit` another, I get the following error:

```
NoMethodError: undefined method `dispose' for #<Shoes::Swt::ArrowPainter:0x5f9be66c>
              dispose at /Users/x/.rvm/gems/jruby-9.1.6.0/gems/shoes-swt-4.0.0.pre9/lib/shoes/swt/arrow.rb:32
               remove at /Users/x/.rvm/gems/jruby-9.1.6.0/gems/shoes-swt-4.0.0.pre9/lib/shoes/swt/common/remove.rb:11
               remove at /Users/x/.rvm/gems/jruby-9.1.6.0/gems/shoes-core-4.0.0.pre9/lib/shoes/common/remove.rb:7
      block in remove at /Users/x/.rvm/gems/jruby-9.1.6.0/gems/after_do-0.4.0/lib/after_do.rb:99
         reverse_each at org/jruby/RubyArray.java:1818
                clear at /Users/x/.rvm/gems/jruby-9.1.6.0/gems/shoes-core-4.0.0.pre9/lib/shoes/slot_contents.rb:35
                clear at /Users/x/.rvm/gems/jruby-9.1.6.0/gems/shoes-core-4.0.0.pre9/lib/shoes/slot.rb:46
                clear at /Users/x/.rvm/gems/jruby-9.1.6.0/gems/shoes-core-4.0.0.pre9/lib/shoes/internal_app.rb:64
                clear at /Users/x/.rvm/rubies/jruby-9.1.6.0/lib/ruby/stdlib/forwardable.rb:189
                visit at /Users/x/.rvm/gems/jruby-9.1.6.0/gems/shoes-core-4.0.0.pre9/lib/shoes/dsl/interaction.rb:45
```

**Comment by jasonrclark** [comment]
Thanks for the bug report @bigonese! Any chance you've got a sample you could share duplicating it? I'm betting I can reconstruct one relatively easily, but I'm moving at the moment, so anything that cuts down the time to repro makes it likely I can get to it sooner.

**Comment by bigonese** [comment]
I don't have a sample I can share readily available, but I'll see if I can pull one together. If it helps in the mean time, I do think I've got a fix ([see this commit](https://github.com/bigonese/shoes4/commit/6db2cf63dcd1ccde4a791db6b33d0f15dbe8b722)), I just haven't had time to test it.

**Comment by bigonese** [comment]
Oops, sorry! That commit was to fix a separate bug that I haven't added a bug report for, yet.

**Comment by jasonrclark** [comment]
Fix incoming on this with #1456
